src/config-generator.js

```diff
@@ -10,22 +10,33 @@ import {
   degenType,
 } from './generator.js'
 
+export type ConfigGenerator = {
+  exports: {[string]: string},
+  inputFile: string,
+  outputFile: string,
+}
+
 export type Config<CustomType: string, CustomImport: string> = {
   baseDir: string,
   generatedPreamble: string,
   // TODO: Support tuples.
-  generators: Array<Array<string>>,
+  generators: Array<ConfigGenerator>,
   importLocations: {[CustomImport]: string},
   typeLocations: {[CustomType]: string},
 }
 
 const stringType = { name: 'string', typeParams: [] }
 const configType = { name: 'Config', typeParams: [ stringType, stringType ]}
+const configGeneratorType = { name: 'ConfigGenerator', typeParams: []}
 
-export default () => degenObject<string, string>(configType, [
+export const degenConfig = () => degenObject<string, string>(configType, [
   degenField('baseDir', degenString()),
   degenField('generatedPreamble', degenString()),
   degenField('typeLocations', degenMapping(degenString(), degenString())),
   degenField('importLocations', degenMapping(degenString(), degenString())),
-  degenField('generators', degenList(degenList(degenFilePath()))),
+  degenField('generators', degenList(degenObject(configGeneratorType, [
+    degenField('exports', degenMapping(degenString(), degenString())),
+    degenField('inputFile', degenString()),
+    degenField('outputFile', degenString()),
```

Comment:
This may just be a minor item, since `degenFilePath` is the same as `degenString`, but if in the future `degenFilePath` is smart about refining paths, there will have been a loss of specificity in the spec.

Reply:
Changed to use `degenFilePath` where appropriate!